FIX improve legacy_expression_tool.md documentation (#1410)
telefonicaid/iotagent-node-lib: scripts/legacy_expression_tool/README.md

* `ignore`: changes expressions from legacy to JEXL equivalence in the [fields where expressions may be used](#fields-in-which-expressions-may-be-used) (based on the translation dictionary specified by `--translations`). In addition, it leaves untouched the `expressionLanguage` field. This may cause inconsistencies, if the value of the `expressionLanguage` is `legacy`, as detailed [in this section](#replacing-expression-without-setting-jexl-at-group-or-device-level).
* `jexl`: changes expressions from legacy to JEXL equivalence in the [fields where expressions may be used](#fields-in-which-expressions-may-be-used) (based on the translation dictionary specified by `--translations`). In addition, it set `expressionLanguage` field to `jexl` (no matter if the field originally exists in the group/device or not) **if some JEXL expression were translated**.

> * `jexlall`: changes expression from legacy to JEXL equivalence in the [fields where expressions may be used](#fields-in-which-expressions-may-be-used) (based on the translation dictionary specified by `--translations`). In addition, it set `expressionLanguage` field to `jexl` (no matter if the field originally exists in the group/device or not) **no matter if some JEXL expression were translated or not**.

Reviewer: ```suggestion
* `jexlall`: changes expression from legacy to JEXL equivalence in the [fields where expressions may be used](#fields-in-which-expressions-may-be-used) (based on the translation dictionary specified by `--translations`). In addition, it set `expressionLanguage` field to `jexl` (no matter if the field originally exists in the group/device or not). The difference between `jexl`and `jexlall` is in the second case, the script is not only looking for documents that contains legacy expressions, it also includes all groups that have `expressionLanguage` field defined. The `expressionLanguage` field on those documents (and also on the documents that contains legacy expresions) is set to `jexl`.
```

> The different between `jexl` and `jexall` is that in the former case the `expressionLanuage` field is set to `jexl` only if some expression were translated and the latter sets the field always, no matter if some expression were translated or not.
> 

Author: ```suggestion
```

#### Execution with `expressionlanguage` set to `jexlall`

When executing the script with `expressionlanguage` set to `jexlall`, the script will look for all the documents

> containing legacy expressions (in some of the expression capable fields) or the existence of the `expressionLanguage` field.

Author: It would be great if the list of "expression capable fields" could be added here.

Reviewer: The script looks for expressions in the following fields:

- active.expression
- active.entity_name
- active.reverse
- attributes.expression
- attributes.entity_name
- attributes.expression
- commands.expression
- endpoint
- entityNameExp
- explicitAttrs

This would change the number of documents found, and the statistics will include extra documents.

> Running the script with the option set to `jexl` would not include such extra documents in statistics.

Author: "statistics will be different" or "impact of statistics" doesn't describe the exact differences or impact.

My suggested wording try to be more precise, although I don't know if I'm fully right.

Please improve if needed.

Reviewer: You are right. I agree the rewording describes better that particular scenario.

`expressionLanguage` field will still be set to the default value or legacy. This would make expression evaluation to
fail, propagating the value of the attribute as the expression literal to the context broker.

> In sum, it is recommented use always the `--expressionlangauge` parameter (note that is't optional) and doing it with a value different from `ignore`.

Author: I have tried to simplify this, replacing a paragraph a bit hard to follow (at least for me :) by a simple statement saying it's a bad idead to use `ignore`.

Reviewer: It is better to suggest `jexl` value for `--expresionlanguage`. As an example, using `delete` option would not help.

```suggestion
To avoid this, it is recommented use always the `--expressionlangauge` parameter set to `jexl` (note that is't optional) and doing it with a value different from `ignore`.
```

> In sum, it is recommented use always the `--expressionlangauge` parameter (note that is't optional) and doing it with a value different from `ignore`.

Reviewer: It is better to suggest `jexl` value for `--expresionlanguage`. As an example, using `delete` option would not help.

```suggestion
To avoid this, it is recommented use always the `--expressionlangauge` parameter set to `jexl` (note that is't optional) and doing it with a value different from `ignore`.
```

> To avoid this, it is recommented use always the `--expressionlangauge` parameter set to `jexl` (note that is't optional) and doing it with a value different from `ignore`.

Author: ```suggestion
To avoid this, it is recommended use always the `--expressionlangauge` parameter set to `jexl`, so doing it a value different from `ignore` will be used.
```
